packages/api/src/util/getMetadata.ts

```diff
@@ -0,0 +1,44 @@
+import { getProvider } from '@cennznet/api/util/getProvider';
+import { TypeRegistry } from '@polkadot/types/create';
+import { RpcCore } from '@polkadot/rpc-core';
+import { MetadataVersioned } from '@polkadot/metadata/MetadataVersioned';
+import { Metadata } from '@polkadot/metadata';
+import { ProviderInterface } from '@polkadot/rpc-provider/types';
+import { WsProvider } from '@polkadot/rpc-provider';
+
+const KEEP = ['System', 'Timestamp', 'TransactionPayment', 'GenericAsset', 'Cennzx'];
+
+export async function getMetadata(
+  provider: string | ProviderInterface,
+  keepMetaFor: string[] = KEEP
+): Promise<Record<string, string>> {
+  let providerInterface: ProviderInterface;
+  if (typeof provider === 'string') {
+    providerInterface = getProvider(provider);
+  } else {
+    providerInterface = provider || new WsProvider();
+  }
+  const registry = new TypeRegistry();
+  const rpcCore = new RpcCore('1', registry, providerInterface);
+  const meta: Metadata = await rpcCore.state.getMetadata().toPromise();
+  const magicNumber = meta.magicNumber;
+  const modules = meta.asLatest.modules.toArray();
+
+  const extrinsic = meta.asLatest.extrinsic;
+  const filteredModule = modules.filter((mod) => keepMetaFor.find((a) => a === mod.name.toString()));
```

Comment:
we should do something nicer here like provide constants or normalise the string so that there's no confusion for users if they write 'system' or 'System' or 'System '

Reply:
added this change